ts/types/wallet.ts

```diff
@@ -1,91 +1,63 @@
-import { ImageSource } from "react-native-vector-icons/Icon";
+import { fromEither, fromNullable } from "fp-ts/lib/Option";
+import {
+  AmountInEuroCents,
+  AmountInEuroCentsFromNumber
+} from "italia-ts-commons/lib/pagopa";
+import { Amount } from "../../definitions/pagopa/Amount";
+import { Psp } from "../../definitions/pagopa/Psp";
+import { Transaction } from "../../definitions/pagopa/Transaction";
+import {
+  UNKNOWN_AMOUNT,
+  UNKNOWN_DATE,
+  UNKNOWN_NUMBER,
+  UNKNOWN_PAYMENT_REASON,
+  UNKNOWN_STRING
+} from "./unknown";
+
+const extractAmount = (amountObject: Amount | undefined): AmountInEuroCents =>
+  // tslint:disable-next-line no-inferred-empty-object-type
+  fromNullable(amountObject) // amountObject.amount is an Amount, which is defined as { amount?: number, ... }
+    .mapNullable(a => a.amount) // if a.amount (i.e. amountObject.amount.amount) is undefined => None
+    .chain(
+      // if the value was extracted, map it to an AmountInEuroCents
+      amount => fromEither(AmountInEuroCentsFromNumber.decode(amount / 100)) // the amount in Amount is stored in cents by pagoPA
+    )
+    .getOrElse(UNKNOWN_AMOUNT); // if anything went wrong, revert to UNKNOWN_AMOUNT
 
 /**
- * Definition of other types required
- * by the app
- * TODO:
- *  - these types may need to be aligned with the PagoPA ones
- *      @https://www.pivotaltracker.com/story/show/157769657
- *  -  verify if email, phone numbers etc can became responsive
- *      @https://www.pivotaltracker.com/n/projects/2048617/stories/158330111
+ * Transaction extractors
  */
 
-/** This type represents the transaction as reported in the notice sent to the user; it contains the data
- *  used by the app to identify the transation the user wants to perform.
- */
-export type NotifiedTransaction = Readonly<{
-  noticeCode: string;
-  notifiedAmount: number;
-  currentAmount: number; // also in WalletTransaction
-  expireDate: Date;
-  tranche: string;
-  paymentReason: string; // also in WalletTransaction
-  cbill: string;
-  iuv: string;
-}>;
+export const getTransactionCreationDate = (t: Transaction): Date =>
```

Comment:
these "getters" are used to conceal the internal structure of `Transaction`s from the rest of the code (mainly to avoid having to interact with the missing values that might result from the "all optional properties" in the pagoPA swagger)

I will probably split these into 3 separate files (transaction.ts, wallet.ts, psp.ts) with the specific getters. @cloudify how about something like:
```
types/
    getters/
        wallet.ts
        transaction.ts
        psp.ts
```
?

Reply:
I'm not sure it makes sense to check every property for nullity and fallback to a default string, perhaps we could simplify it and define a minimum subset of properties that are required, if any of those is null, we skip the object as invalid - what do you think about his?

Reply:
But, in this case, we would have to build and maintain a separate type based on the pagoPA one, meaning that we would miss out on the benefits that come with automatically building the definitions from the swagger. 

I have already pointed out this "optional-only" flaw in the swagger definitions and I'm hoping the pagoPA team decides to consider fixing it. If that happens, this code can be updated fairly easily. 

Would you reccommend building new types anyway?

Reply:
About this, I'm having problems with the static typing. The following is an example I would expect to fail at transpile time.
```typescript
const TypeWithNullableProp = t.partial({
  numberOrUndefined: t.number
});
type TypeWithNullableProp = t.TypeOf<typeof TypeWithNullableProp>;

const TypeWithRequiredProp = t.refinement(
  TypeWithNullableProp,
  a => a.numberOrUndefined !== undefined
);
type TypeWithRequiredProp = t.TypeOf<typeof TypeWithRequiredProp>;

const test: TypeWithNullableProp = {}; // ok
const test2: TypeWithRequiredProp = {}; // would expect this to trigger an error
```
Without this, creating a refinement of the original type is "only" useful at runtime (which is great already), but the bulk of the code would still require handling undefined cases as is currently being done with the getters

Reply:
yes, unfortunately the predicate is for runtime checks, `refinement` [doesn't infer the new type from the predicate](https://github.com/gcanti/io-ts/blob/master/src/index.ts#L290) - you probably have to create a type as well and use it in place of `t.TypeOf`:

```ts
type TypeWithNullableProp = {
    a?: string;
    b?: string;
    c?: string;
}

type TypeWithRequiredProp = TypeWithNullableProp & Required<Pick< TypeWithNullableProp, "a" | "b">>;

const b0: TypeWithRequiredProp = {} // error

const b1: TypeWithRequiredProp = {
    a: "a",
    b: "b"
};
```